src/Builder/Visitor/AddPluginToPluginCollectionVisitor.php

```diff
@@ -114,4 +134,40 @@ public function enterNode(Node $node): Node
 
         return $node;
     }
+
+    /**
+     * @param \PhpParser\Node $node
+     *
+     * @return array<string>
+     */
+    protected function getPluginList(Node $node): array
+    {
+        $plugins = [];
+
+        foreach ($node->stmts as $stmt) {
+            if (
+                $stmt->expr instanceof MethodCall === false
+                || strpos(strtolower($stmt->expr->name->toString()), 'add') === false
+            ) {
+                continue;
+            }
+
+            /** @var \PhpParser\Node\Arg $arg */
+            foreach ($stmt->expr->args as $arg) {
+                if ($arg->value instanceof New_) {
+                    $plugins[] = $arg->value->class->toString();
+                }
+            }
+        }
+
+        return $plugins;
+    }
+
+    /**
+     * @return \SprykerSdk\Integrator\Builder\Visitor\PluginPositionResolver\PluginPositionResolverInterface
+     */
+    protected function getPluginPositionResolver(): PluginPositionResolverInterface
+    {
+        return new PluginPositionResolver();
+    }
```

Comment:
Here is the same. Use DI in constructor for PluginPositionResolver

Reply:
++